lib/build.ts

```diff
@@ -109,16 +109,16 @@ export async function runBuildTask(
 			'>=1.38.0',
 		);
 
-	task.dockerOpts = {
+	task.dockerOpts = _.merge(
```

Comment:
In PR #85, I incorrectly advised that lodash's `_.merge` could be replaced with the `...` spread syntax, but there are corner cases:
* If `task.dockerOpts` contains properties set to `undefined`, `_.merge` skips them while the spread syntax sets the option to `undefined`. For example:
```
> _.merge({ a: 1 }, { a: undefined }, undefined, { b: 2 })
{ a: 1, b: 2 }

> { a: 1, ...{ a: undefined } }
{ a: undefined }
```

* `_.merge` recursively merges objects in depth, so that `registryconfig` values would be added together rather than replaced:

```
> _.merge({ o: { b: 2 } }, { o: { c: 3 } })
{ o: { b: 2, c: 3 } }

> {o: { b: 2 }, ...{ o: { c: 3 } } }
{ o: { c: 3 } }
```

Reply:
Good catch! could we add a test so that we are sure that this doesn't happen again in a future change? 

There's a trend towards dropping the use of lodash functions in Balena... and given how we work, it could happen that some future developer replaces this back to the spread syntax. A test could prevent this :+1:

Reply:
Thanks. I've modified an existing test in `build.spec.ts` to cover the case of `task.dockerOpts` containing a `platform: undefined` value.